### PR TITLE
market roles tests

### PR DESCRIPTION
**Pull request by xnetsrak**
## Description

init marketrole component tests added

## References

- #0000

### CODE REVIEW

File: libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/dh-market-participant-actor-market-roles.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        DhApiModule.forRoot(),
+        getTranslocoTestingModule(),
+      ],
+    });
+
+    await view.fixture.whenStable();
+
+    view.fixture.detectChanges();
+
+    const allOptions = screen.getAllByRole('option');
```
Comment (frontend-specialisten) [inline libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/dh-market-participant-actor-market-roles.component.spec.ts:81]: Nitpick: common queries can also be extracted to provide reduce duplications but still keeping the readability. This example could be `getAllOptions = () => screen.getAllByRole('option')` which returns the query.

**Review by frontend-specialisten — CHANGES_REQUESTED** [review]
Looks good, only a single `suggestion` and some `nitpicks` you can consider :)

**Review by frontend-specialisten — APPROVED** [review]

File: libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/dh-market-participant-actor-market-roles.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    const selectedRoles = allOptions.filter(
+      (e) => e.attributes.getNamedItem('aria-selected')?.value == 'true'
+    );
+    const changes = view.fixture.componentInstance.changes;
+
+    expect(selectedRoles).toHaveLength([elemToSelect1, elemToSelect2].length);
+    expect(changes.marketRoles).toHaveLength(
+      [elemToSelect1, elemToSelect2].length
+    );
+  });
```
Comment (dzhavat) [inline libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/dh-market-participant-actor-market-roles.component.spec.ts:119]: suggestion: it'd be nice to have at least one test that covers the output functionality. It's an important functionality for this component.

File: libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/dh-market-participant-actor-market-roles.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import { DhMarketParticipantActorMarketRolesComponent } from './dh-market-participant-actor-market-roles.component';
+import { getTranslocoTestingModule } from '@energinet-datahub/dh/shared/test-util-i18n';
+import { DhApiModule } from '@energinet-datahub/dh/shared/data-access-api';
+import { MarketRoleService } from './market-role.service';
+import {
+  EicFunction,
+  MarketRoleDto,
+} from '@energinet-datahub/dh/shared/domain';
+
+describe('MarketRolesChk', () => {
```
[the remark this answers is not in the document]
Comment (xnetsrak) [reply libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/dh-market-participant-actor-market-roles.component.spec.ts]: tests dit not run on name change

**Pull request closed without merging**